controllers/upgrade_handlers.go

```diff
@@ -76,5 +141,49 @@ func (r *ImageBasedUpgradeReconciler) handleUpgrade(ctx context.Context, ibu *lc
 		metav1.ConditionFalse,
 		"Upgrade completed",
 		ibu.Generation)
+
+	return doNotRequeue(), nil
+}
+
+// handleBackup manages backup flow and returns with possible requeue
+func (r *ImageBasedUpgradeReconciler) handleBackup(ctx context.Context, backupCRs []*velerov1.Backup) (ctrl.Result, error) {
+	// sort backupCRs by wave-apply
+	sortedBackupGroups, err := backuprestore.SortByApplyWaveBackupCrs(backupCRs)
+	if err != nil {
+		return requeueWithError(err)
+	}
+
+	// trigger and track each group
+	for _, backups := range sortedBackupGroups {
+		backupTracker, err := r.BackupRestore.TriggerBackup(ctx, backups)
+		if err != nil {
+			return requeueWithError(err)
+		}
+		if len(backupTracker.SucceededBackups) == len(backups) {
+			// The current backup group has done, work on the next group
+			continue
+		} else if len(backupTracker.FailedBackups) != 0 {
+			// not recoverable
+			msg := fmt.Sprintf("Failed backups: %s", strings.Join(backupTracker.FailedBackups, ","))
+			BackupFailed = fmt.Errorf("%s: %s", BackupFailed.Error(), msg)
+			r.Log.Error(BackupFailed, "")
+			return doNotRequeue(), BackupFailed
+		} else if len(backupTracker.ProgressingBackups) != 0 {
+			r.Log.Info(fmt.Sprintf("Inprogress backups: %s", strings.Join(backupTracker.ProgressingBackups, ",")))
+			return requeueWithCustomInterval(2 * time.Second), nil
+		} else if len(backupTracker.FailedValidationBackups) != 0 {
+			msg := fmt.Sprintf("Failed validation backups reported by Oadp: %s. %s", strings.Join(backupTracker.FailedValidationBackups, ","), "Please update the invalid backups.")
+			BackupValidationFailedShortRequeue = fmt.Errorf(msg)
```

Comment:
This is defined as gobal variable on the top. Will that be overriden?

Reply:
Updated implementation to make it clearer